ui/components/app/assets/token-list/token-list.tsx

```diff
@@ -1,56 +1,105 @@
-import React, { ReactNode, useEffect } from 'react';
-import { useSelector } from 'react-redux';
+import React, { useEffect, useMemo } from 'react';
+import { shallowEqual, useSelector } from 'react-redux';
 import { Hex } from '@metamask/utils';
 import TokenCell from '../token-cell';
-import { getChainIdsToPoll, getPreferences } from '../../../../selectors';
+import {
+  getChainIdsToPoll,
+  getCurrentNetwork,
+  getNewTokensImported,
+  getPreferences,
+  getSelectedAccount,
+  getTokenBalancesEvm,
+  getTokenExchangeRates,
+} from '../../../../selectors';
 import { endTrace, TraceName } from '../../../../../shared/lib/trace';
 import { useTokenBalances as pollAndUpdateEvmBalances } from '../../../../hooks/useTokenBalances';
-import useSortedFilteredTokens from '../hooks/useSortedFilteredTokens';
-import useShouldShowFiat from '../hooks/useShouldShowFiat';
+import { useNativeTokenBalance, useNetworkFilter } from '../hooks';
+import { TokenWithFiatAmount } from '../types';
+import { getMultichainIsEvm } from '../../../../selectors/multichain';
+import { getConversionRate } from '../../../../ducks/metamask/metamask';
+import { filterAssets } from '../util/filter';
+import { sortAssets } from '../util/sort';
 
 type TokenListProps = {
   onTokenClick: (chainId: string, address: string) => void;
-  nativeToken?: ReactNode;
 };
 
-export default function TokenList({
-  onTokenClick,
-  nativeToken,
-}: TokenListProps) {
-  const { privacyMode } = useSelector(getPreferences);
+export default function TokenList({ onTokenClick }: TokenListProps) {
+  const isEvm = useSelector(getMultichainIsEvm);
   const chainIdsToPoll = useSelector(getChainIdsToPoll);
+  const newTokensImported = useSelector(getNewTokensImported);
+  const evmBalances = useSelector(getTokenBalancesEvm); // TODO: This is where we need to select non evm-assets from state, when isEvm is false
+  const currentNetwork = useSelector(getCurrentNetwork);
+  const { tokenSortConfig, privacyMode } = useSelector(getPreferences);
+  const selectedAccount = useSelector(getSelectedAccount);
+  const conversionRate = useSelector(getConversionRate);
+  const contractExchangeRates = useSelector(
+    getTokenExchangeRates,
+    shallowEqual,
+  );
 
   // EVM specific tokenBalance polling, updates state via polling loop per chainId
   pollAndUpdateEvmBalances({
     chainIds: chainIdsToPoll as Hex[],
   });
 
-  const sortedFilteredTokens = useSortedFilteredTokens();
```

Comment:
Felt like there was no need to put this into it's own hook, since it's only being used here.